src/msw/overlay.cpp

```diff
@@ -0,0 +1,227 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/msw/overlay.cpp
+// Author:      Ali Kettab
+// Copyright:   (c) 2022 Ali Kettab
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#include "wx/wxprec.h"
+
+#include "wx/private/overlay.h"
+#include "wx/dcclient.h"
+#include "wx/nativewin.h"
+#include "wx/window.h"
+#include "wx/sysopt.h"
+#include "wx/msw/private.h"
+
+namespace // anonymous
+{
+static const wxChar* gs_overlayClassName = nullptr;
+
+class wxOverlayWindow : public wxNativeContainerWindow
+{
+public:
+    static wxWindow*  Create(bool fullScreen = false)
+    {
+        static const wxChar* OVERLAY_WINDOW_CLASS = wxS("wxOverlayWindow");
```

Comment:
Could you please add a comment explaining why do we need a separate window class for this window? It doesn't seem obvious to me...

Reply:
I first added this for debugging purposes. but when I tried to remove it I get an instant crash!

Reply:
Err, this is rather worrisome because I don't see why should it crash at all. Do you have any information about this crash, i.e. a stack trace or something?

Reply:
It crashes because `CreateWindowEx()` refuses to create the window if `lpClassName` is **null** and the error message returned is **invalid parameter value**. I don't know why, but I also don't mind passing a non-null `lpClassName` if that makes `CreateWindowEx()` happy in the end.

Reply:
Yes, sure, this makes sense, you need to pass a valid class name to `CreateWindow()`.

The question is why do we need a specific class name and can't just use `wxWindowNR` here (this is returned by `wxWindow::GetMSWClassName()`)?

Reply:
I tried to do this but whenever the overlay window is shown for the first time there is a noticeable blinking on the window underneath (may be due to a window refresh ?) which is definitely undesirable/unacceptable.